tests/sentry/api/endpoints/test_project_details.py

```diff
@@ -194,6 +196,104 @@ def test_non_org_rename_403(self):
         self.get_error_response(other_org.slug, "old_slug", status_code=403)
 
 
+class ProjectUpdateTestTokenAuthenticated(APITestCase):
+    endpoint = "sentry-api-0-project-details"
+    method = "put"
+
+    def setUp(self):
+        super().setUp()
+        self.org_slug = self.project.organization.slug
+        self.proj_slug = self.project.slug
+
+    def test_member_updates_denied_with_token(self):
+        project = self.create_project(platform="javascript")
```

Comment:
In all these tests, you create the same project, user, and member. Can you move this set up to `setUp`?

Reply:
Done! I've kept the `create_member` call in each of the tests however since some of them have the `member` role while others have `admin`.